app/components/primer/auto_complete.rb

```diff
@@ -1,98 +1,150 @@
 # frozen_string_literal: true
 
 module Primer
-  # Use `AutoComplete` to populate input values from server search results.
+  # Use `AutoComplete` to provide a user with a list of selectable suggestions that appear when they type into the
+  # input field. This list is populated by server search results.
   # @accessibility
-  #   Always provide a label for the `Autocomplete` component.
+  #   Always set an accessible label to help the user interact with the component.
+  #
+  #   * Set the `label` slot to render a visible label.
+  #   * If you must use a non-visible label, set `:"aria-label"` on `AutoComplete` and Primer
+  #   will apply it to the correct elements. However, please note that a visible label should almost
+  #   always be used unless there is compelling reason not to. A placeholder is not a label.
   class AutoComplete < Primer::Component
     status :beta
 
-    DEFAULT_INPUT_TYPE = :text
-    INPUT_TYPE_OPTIONS = [DEFAULT_INPUT_TYPE, :search].freeze
+    # Optionally render a visible label. See <%= link_to_accessibility %>
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :label, lambda { |**system_arguments|
+      system_arguments[:for] = @input_id
+      system_arguments[:tag] = :label
+      Primer::BaseComponent.new(**system_arguments)
+    }
 
     # Required input used to search for results
     #
-    # @param type [Symbol] <%= one_of(Primer::AutoComplete::INPUT_TYPE_OPTIONS) %>
+    # @param type [Symbol] <%= one_of(Primer::AutoComplete::Input::TYPE_OPTIONS) %>
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-    renders_one :input, lambda { |type: DEFAULT_INPUT_TYPE, classes: "form-control", **system_arguments|
-      system_arguments[:tag] = :input
-      system_arguments[:type] = fetch_or_fallback(INPUT_TYPE_OPTIONS, type, DEFAULT_INPUT_TYPE)
-      system_arguments[:classes] = classes
-      Primer::BaseComponent.new(**system_arguments)
+    renders_one :input, lambda { |**system_arguments|
+      aria_label = system_arguments[:"aria-label"] || system_arguments.dig(:aria, :label) || @aria_label
+      if aria_label.present?
+        system_arguments[:"aria-label"] = aria_label
+        system_arguments[:aria]&.delete(:label)
+      end
+
+      Input.new(id: @input_id, **system_arguments)
     }
 
+    # This component is part of `Primer::AutoCompleteComponent` and should not be
+    # used as a standalone component.
+    class Input < Primer::Component
```

Comment:
I'd prefer if this class was on the bottom, it gets hard to read having a class in the middle of another class